src/mame/drivers/geneve.cpp

```diff
@@ -434,7 +431,7 @@ WRITE8_MEMBER ( geneve_state::cruwrite )
 READ8_MEMBER( geneve_state::cruread )
 {
 	uint8_t value = 0;
-	int addroff = offset << 4;
+	uint16_t addroff = offset << 1;
```

Comment:
Since we are using the software addressing, the <<1 is wrong here.

Reply:
Perhaps the emulation of the various CRU buses should switch to using the hardware address, since that's what both read and write handlers will see now.